### PR TITLE
Add server for Hadoop RPC.

### PR DESCRIPTION
**Pull request by pelmers**
The server resides in rpcserver.
Attach handlers via register_handler, an example lives in rice-namenode/main.cc.

Try testing by running the namenode, then in another window execute
`hdfs dfs -fs hdfs://localhost:<port> -mkdir foo`

r? anyone who'd like to

### CODE REVIEW

File: rice-namenode/CMakeLists.txt [hunk 1]
```diff
@@ -1 +1,8 @@
+include_directories(${CMAKE_SOURCE_DIR}/rpcserver)
+
 add_executable(namenode main.cc)
+target_link_libraries( namenode rpcserver )
```
Comment (efrederic) [inline rice-namenode/CMakeLists.txt:4]: Just for the sake of following a consistent style, we should get rid of the spaces inside these parens (lines 4-8)

**Review by efrederic — APPROVED** [review]

File: rpcserver/socket_writes.cc [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+using asio::ip::tcp;
+
+namespace rpcserver {
+
+    /**
+     * Return whether an attempt to write val on socket is successful.
+     */
+    bool write_int32(tcp::socket& sock, uint32_t val) {
+        asio::error_code error;
+        size_t write_len = sock.write_some(asio::buffer(&val, 4), error);
```
Comment (srdieffenbach) [inline rpcserver/socket_writes.cc]: Should htonl be called first, as is done symmetrically in socket_reads:read_int32?

**Review by srdieffenbach — COMMENTED** [review]
Comment on socket_writes:write_in32

File: rpcserver/socket_writes.cc (hunk 2 above)
Comment (pelmers) [reply rpcserver/socket_writes.cc]: Yep, good catch; I hadn't noticed it because the stub message has length 0 >_<

**Comment by jfking** [comment]
Hey @pelmers remember to move this along on the project board

**Pull request merged**